src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -68,23 +73,54 @@ const CustomFieldInputSingleChoice = forwardRef(function CustomFieldInputSingleC
     props.onChange({ target: selfRef.current });
   }, [value]);
 
+  useEffect(() => {
+    const getChoices = async () => {
+      await execute(`${API_ROOT}/custom_field_choices?for_custom_fields=${props.id}`)
+        .then(({ json, mounted }) => {
+          if (mounted) {
+            const mungedChoices = json.results.map((result) => {
+              return {
```

Comment:
```suggestion
              return json[result.key][result.id];
```

I think it is preferable to keep using the API object as a whole instead of slicing what we need. What do you think?